src/gui/widgets/FxLine.cpp

```diff
@@ -395,3 +439,39 @@ void FxLine::setStrokeInnerInactive( const QColor & c )
 {
 	m_strokeInnerInactive = c;
 }
+
+
+// Ask user for a color, and set it as the mixer line color
+void FxLine::changeColor()
+{
+	auto channel = Engine::fxMixer()->effectChannel( m_channelIndex );
+	
+	auto new_color = ColorChooser( this ).withPalette( ColorChooser::CCPalette::Mixer )->getColor( channel->m_color );
+	if( ! new_color.isValid() )
+	{ return; }
+	
+	channel->m_color = new_color;
+	channel->m_hasColor = true;
+	
+	update();
+}
+
+
+// Disable the usage of color on this mixer line
+void FxLine::resetColor()
+{
+	Engine::fxMixer()->effectChannel( m_channelIndex )->m_hasColor = false;
+	update();
+}
+
+
+// Pick a random color from the mixer palette and set it as our color
+void FxLine::randomColor()
+{
+	auto channel = Engine::fxMixer()->effectChannel( m_channelIndex );
+	
+	channel->m_color = ColorChooser( this ).withPalette( ColorChooser::CCPalette::Mixer )->standardColor( rand() % 48 );
```

Comment:
I've made it so a palette can be accessed without initialising a ColorDialog, do you think that will help?

Reply:
You should be able to call nicePalette directly, no?

Reply:
No, it's a private function.

Reply:
Yes, but you could make it public. It has no side effects so it's completely safe to make it a public static function with const arguments.

Reply:
A small comment on const arguments: if you're passing them by value, making them const is superfluous. There may be an argument for doing so in function definitions, since making variables const by default is good practice, but in a function declaration, it makes no difference to the function type and only adds noise. I don't tend to mention it in reviews since it's harmless, but it's something that's generally avoided.

Reply:
@Spekular, considering that there's a way to access the palette now, do you think that is enough or do you still want to directly access the algorithm?

@DomClark, as far as I've understood, you meant to say the argument having a `const` prefix is not required in this context, so I removed the `const`. Is that what you meant for me to do?

Reply:
> @Spekular, considering that there's a way to access the palette now, do you think that is enough or do you still want to directly access the algorithm?

As mentioned on Discord, I prefer using getPalette over direct access, wasn't thinking properly earlier. Will fix my review ASAP.

Reply:
You don't have to construct `ColorPalette` if you make `getPalette()` static.

Reply:
Alright, updated. Did I get it right?